### PR TITLE
touchring and cursor are now two modules, added better cursor, fixed positioning of cursor (depends on image aswell)

### PR DESCRIPTION
**Pull request by fscz**

### CODE REVIEW

**Comment by tito** [comment]
I'll like to delay this one until 1.9 is out :) Touchring itself is impacted. Cursor module is a good one :)

**Comment by tshirtman** [comment]
may i ask where the cursor image is from, and what's its license? it's probably trivial to find one with a liberal enough license, but we must check :).

File: kivy/modules/cursor.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+Displays an artistic cursor texture pointing at the position of the mouse cursor. This is practical for use with the egl_rpi window that is drawn right into the framebuffer and has no cursor by default.
+
+Configuration
+-------------
+
+:Parameters:
+    `cursor_size`: tuple, defaults to (40, 40)
+        Apparent size of the mouse cursor, if displayed, (None,None) value
+        will keep its real size.
+        .. versionadded:: 1.9.0
```
Comment (akshayaurora) [inline kivy/modules/cursor.py]: 1.9.1

File: kivy/modules/cursor.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    `cursor_offset`: tuple, defaults to (None, None)
+        Offset of the texture image. The default value will align the
+        top-left corner of the image to the mouse pos.
+        .. versionadded:: 1.9.0
```
Comment (akshayaurora) [inline kivy/modules/cursor.py]: 1.9.1

**Comment by tshirtman** [comment]
also, had a few issues raining it (+ some style issues), here is a patch to it

https://gist.github.com/10a70a804b49d33f7681

did it work for you with (40,40) notation? (instead of 40x40)

**Comment by KeyWeeUsr** [comment]
@fscz your repo does not exist. Rebase/close or someone wants it? And still no info about cursor license(search `cursor "cc0"` and replace it if the image is bad) ^^

**Comment by tshirtman** [comment]
closing in favor of #4721

**Pull request closed without merging**